README.md

```diff
@@ -174,6 +175,33 @@ curl --request GET 'http://localhost:4000/api/checks/executions/205e326d-0c25-4f
 
 Refer to the [API doc](http://localhost:4000/swaggerui) for more information about requests and responses.
 
+#### **Debugging gathered facts**
+
+Often times knowing the returned value of the gathered facts is not a trivial thing, more during the implementation of new checks.
+
+To better debug the fact gathering process and the returned values the `facts` subcommand of `trento-agent` is a really useful tool. This command helps to see in the target itself what the gathered fact looks like. This is specially interesting when the returned value is a complex object or the target under test is modified and the check developer wants to see how this affects to the gathered fact.
+
+The command can be used as:
+
+```
+./trento-agent facts gather --gatherer corosync.conf --argument totem.token
+# To see the currently available gatherers and their names
+# ./trento-agent facts list
+```
+
+Which would return the next where the `Value` is the available value in the written check:
+
+```
+{
+  "Name": "totem.token",
+  "CheckID": "",
+  "Value": {
+    "Value": 30000
```

Comment:
Do you think it is worth adding a note to clarify the double `Value`?
It's tricky though, because it carries serialization concerns which might be out of scope for this piece of doc.

Maybe we could just add a reference to the relation with the DSL, for instance just saying that there's a serialization/transformation process that allows to access the fact like `facts.<fact_name>` and get back `30000`.

To make it more realistic we could chose another fact name for this example because `totem.token` is more of an argument and I don't know whether we support, or even if we should support, dotted fact names like `totem.token`.

Reply:
We cannot change the fact name, this is how the `facts` subcommand works. It puts the passed argument as name

Reply:
I see. Maybe it could just be fixed to `fact_name` or similar :smile: in the end this is a debugging tool.
However, I am also fine with how it is now.

Reply:
Maybe we should update the `facts` subcommand in the agent side, and add a `--json` flag or something to print something readable, and not the internal `FactValue` stuff

Reply:
tracked as debt, we could even try to output rhai syntax